skimage/restoration/_nl_means_denoising.pyx

```diff
@@ -80,16 +26,16 @@ cdef inline double patch_distance_2dmultichannel(IMGDTYPE [:, :, :] p1,
         Second patch, 2D image with last dimension corresponding to channels.
     w : 2-D array_like
         Array of weights for the different pixels of the patches.
-    s : int
+    s : Py_ssize_t
         Linear size of the patches.
-    var : double
+    var : np_floats
```

Comment:
I see that you moved the `2 * var` outside of this function for efficiency. That seems fine, but I would update the docstring here and possibly the variable name to indicate that this is not actually the variance of the noise anymore.
Maybe use a name something like `var_diff`, indicating that `var_diff == 2 * var` (i.e. var_diff is the variance of the subtraction of two random variables, each with variance `var`)